[a11y] Form a11y docs (#1636)
Shopify/polaris-react: src/components/Form/README.md

- To arrange fields within a form using standard spacing, [use the form layout component](/components/forms/form-layout)
- To see all of the components that make up a form, [visit the form section](/components/forms/checkbox#navigation) of the component library

---

## Accessibility

<!-- content-for: android -->

See Material Design and development documentation about accessibility for Android:

- [Accessible design on Android](https://material.io/design/usability/accessibility.html)
- [Accessible development on Android](https://developer.android.com/guide/topics/ui/accessibility/)

<!-- /content-for -->

<!-- content-for: ios -->

See Apple’s Human Interface Guidelines and API documentation about accessibility for iOS:

- [Accessible design on iOS](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/accessibility/)
- [Accessible development on iOS](https://developer.apple.com/accessibility/ios/)

<!-- /content-for -->

<!-- content-for: web -->

The form component wraps content in an HTML `<form>` element. This helps to support assistive technologies that use different interaction and browse modes.

> By default, any buttons that you add to the form are given a `type` attribute set to `button`. Use the `submit` prop to set the `type` attribute to `submit` instead. The form can have only one submit button, and it must be at the end of the form.

Reviewer: > Use the `submit` prop to set the `type` attribute to `submit` instead.

Here I'm unclear whether you mean the submit prop on `Button` or `Form` 🤔 

```suggestion
Forms can have only one submit button and it must be at the end of the form. By default, all buttons contained by the form component are given a `type` attribute set to `button`. Set the `submit` prop on the button component to set its `type` attribute to `submit` instead.
```

Author: @chloerice 

Would this be clearer?

>>> Forms can have only one submit button and it must be at the end of the form. By default, all buttons _added to_ the form component are given a `type` attribute set to `button` to avoid conflicts. To make a button the submit button instead, set the `submit` prop on the button.
